src/core/mtu_discovery.h

```diff
@@ -0,0 +1,121 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+--*/
+
+typedef enum QUIC_MTU_DISCOVERY_STATE {
+
+    //
+    // MTU discovery is in the searching state.
+    //
+    QUIC_MTU_DISCOVERY_STATE_SEARCHING,
+
+    //
+    // MTU discovery is in the waiting state.
+    //
+    QUIC_MTU_DISCOVERY_STATE_SEARCH_COMPLETE,
+
+} QUIC_MTU_DISCOVERY_STATE;
```

Comment:
Looks like this can be simplified to a `IsSearching` flag instead.

Reply:
Let's start with simple. No black hole detection. If/when we add that, then we can add more complicated states.